### PR TITLE
Use full qualifyer for images and user quay.io if possible

### PR DESCRIPTION
**Pull request by 6543**
& [bump alpine to alpine:3.17](https://github.com/woodpecker-ci/woodpecker/pull/1659/commits/588f493c47f4660d72ba76c9f1df3ea05f37bbd5)

### CODE REVIEW

File: shared/constant/constant.go [hunk 1]
```diff
@@ -16,10 +16,11 @@ package constant
...
 var PrivilegedPlugins = []string{
-	"plugins/docker",
-	"plugins/gcr",
-	"plugins/ecr",
-	"woodpeckerci/plugin-docker-buildx",
+	"docker.io/plugins/docker",
+	"docker.io/plugins/gcr",
+	"docker.io/plugins/ecr",
+	"docker.io/woodpeckerci/plugin-docker-buildx",
+	"quay.io/woodpeckerci/plugin-docker-buildx",
```
Comment (lafriks) [inline shared/constant/constant.go:23]: maybe add also codeberg?
```suggestion
	"quay.io/woodpeckerci/plugin-docker-buildx",
	"codeberg.org/woodpecker-plugins/docker-buildx",
```

File: .woodpecker/test.yml [hunk 2]
```diff
@@ -46,12 +46,13 @@ pipeline:
...
 
   lint-editorconfig:
-    image: mstruebing/editorconfig-checker
+    image: docker.io/mstruebing/editorconfig-checker
     group: test
 
   securitycheck:
     group: test
-    image: aquasec/trivy:latest
+    pull: true
```
Comment (anbraten) [inline .woodpecker/test.yml:54]: Is this really needed as it creates a lot of traffic?

File: .woodpecker/test.yml [hunk 3]
```diff
@@ -103,7 +104,7 @@ pipeline:
 
   codecov:
     pull: true
```
Comment (anbraten) [inline .woodpecker/test.yml:106]: Should probably be removed as well.

File: .woodpecker/test.yml (hunk 2 above)
Comment (6543) [reply .woodpecker/test.yml:54]: other option would be to add a cron job to keep the latest tagged images up to date ...

Comment (lafriks) [reply .woodpecker/test.yml:54]: Pull won't repull if actual latest image is already cached just check if one exists, while it is some additional traffic I don't think it's that big

**Comment by pat-s** [comment]
Using the FQ for `docker.io `is quite uncommon as its the default anyhow. 

quay for rate limiting or what's the motivation?

**Comment by 6543** [comment]
yes it was created while Docker Inc. try to remove OpenSource orgs

I still think we should still provide an other option as docker-hub. but it's not as urgent anymore

**Comment by 6543** [comment]
we first need ->  #1995

**Comment by pat-s** [comment]
> I still think we should still provide an other option as docker-hub

Nothing to say against this but I think it's not necessary to add `docker.io` everywhere explicitly.

**Comment by xoxys** [comment]
Conflicts resolved in https://github.com/woodpecker-ci/woodpecker/pull/2692

**Pull request closed without merging**